### PR TITLE
`limactl prune`: add `--keep-referred` option to keep objects that are referred by some instances or templates

### PR DESCRIPTION
**Pull request by norio-nomura**
~~Also adds `--all` option to `prune` behaves as same as previous function. (Remove all cacheDir)~~

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Why ? An image cache used by an existing instance can be safely pruned

**Comment by norio-nomura** [comment]
I often run `limactl factory-reset` because it's more convenient for me if the cache is preserved.

**Comment by norio-nomura** [comment]
Cache entries that are not referenced by either the instance or the template are unlikely to be used again. I wanted to delete those.

**Comment by AkihiroSuda** [comment]
Can we just retain the default behavior and  have a new flag like `--keep-referred=<BOOL>`?

**Review by AkihiroSuda — DISMISSED** [review]
Thanks

File: cmd/limactl/prune.go [hunk 1]
```diff
@@ -17,15 +22,115 @@ func newPruneCommand() *cobra.Command {
...
 	}
 	cacheDir := filepath.Join(ucd, "lima")
 	logrus.Infof("Pruning %q", cacheDir)
-	return os.RemoveAll(cacheDir)
+	if !keepReferred {
+		return os.RemoveAll(cacheDir)
+	}
+
+	// Prune downloads that are not used by any instances or templates
+	downloadDir := filepath.Join(cacheDir, "download", "by-url-sha256")
```
Comment (AkihiroSuda) [inline cmd/limactl/prune.go:45]: This internal data structure should be handled in `pkg/downloader` rather than in the CLI

Comment (norio-nomura) [reply cmd/limactl/prune.go]: Moved some codes to `pkg/downloader`.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by norio-nomura** [comment]
Thanks! 🙏🏻

**Pull request merged**